### ISSUE TITLE
[Feature]: Support for multiple column sorting in table widget 

### ISSUE THREAD

**Issue by abhvsn** [appsmithorg/appsmith#9949]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

In the table widget, we can sort using a single column, by enabling the sort for multiple column can be a good feature when we have lots of records in a table and pagination is disabled.   

### Why should this be worked on?

This will improve the visibility for larger dataset and ease the pain to work with these datasets. Today we autopaginate the table after some records but this will definitely be helpful once we have scrollable table widget.

Ref image from user:
![image](https://user-images.githubusercontent.com/41686026/147112558-041ba7e5-e0bb-49f0-a158-01a849e8d754.png)

**Comment by billyrudi** [comment]
i think that this is very useful.
If i can help you on this let me now!

**Comment by dilippitchika** [comment]
Thanks @billyrudi 

@riodeuno something we may consider after adding infinite scrolling table, @somangshu can we discuss this once for the current paginated version?

**Comment by riodeuno** [comment]
@dilippitchika  Sounds good. We need to consider this when designing the infinite scrolling table, and we can implement this in a later phase. 
I am wondering if, the order in which the columns are sorted matters here? Seems like it does.

**Comment by dilippitchika** [comment]
@riodeuno yes, the ideal implementation will need an order to sort. Tagging below an example of how notion does it. 
![Screenshot 2021-12-24 at 11 08 17 AM](https://user-images.githubusercontent.com/19730984/147320924-76bd826b-50c3-4e34-a2a4-ca9c5850a723.png)
t

**Comment by billyrudi** [comment]
it seem good: this will be not at designe time right? i think that the right location is where we have the filter settings.

**Comment by dilippitchika** [comment]
Hey @billyrudi yes, this not at the design time but in the filter settings.
